docs/source/using-discovery/bash.rst

```diff
@@ -0,0 +1,867 @@
+
+Syncing a local directory to a remote server::
+
+   rsync -avz /local/path user@xfer.discovery.neu.edu:/remote/path
+
+Syncing a remote server to a local directory::
+
+   rsync -avz user@xfer.discovery.neu.edu:/remote/path /local/path
+
+Syncing a local directory to a remote server with compression::
+
+   rsync -avz --compress /local/path user@xfer.discovery.neu.edu:/remote/path
+
+Syncing a remote server to a local directory while preserving permissions::
+
+   rsync -avz --perms user@xfer.discovery.neu.edu:/remote/path /local/path
+
+Syncing only files that have been modified in the last hour::
+
+   rsync -avz --update --min-age=3600 /local/path user@xfer.discovery.neu.edu:/remote/path
+
+Syncing a local directory to a remote server while excluding certain files::
+
+   rsync -avz --exclude='*.log' /local/path user@xfer.discovery.neu.edu:/remote/path
+
+Syncing a remote server to a local directory while preserving symbolic links::
+
+   rsync -avz --links user@xfer.discovery.neu.edu:/remote/path /local/path
+
+find
+-----
+``find`` is a command line tool used to search for files and directories within a specified location. It operates by starting at a specified directory and recursively searching through its subdirectories. The user can select a range of criteria to match (e.g., file name, size, modification time), and ``find`` will return a list of all files and directories that match the specified criteria. ``find`` provides a range of options for further processing the results, such as executing a command on each matching file, printing the results, or performing other operations. As a result, it is a versatile tool to search for specific files and to clean up old files.
+
+Here are several advanced examples of using the ``find`` command to search for files and directories; see `find(1) manual page`_ for more information on how to use the command effectively.
+
+To search for files and directories:
+
+.. code-block:: shell
+  :emphasize-lines: 2-3
+
+  find /path/to/search -name "*.txt"
+  /path/to/search/file1.txt
+  /path/to/search/file2.txt
+
+Finding files based on size::
+
+   find /path/to/dir -size +10M
+
+This will find all files in ``/path/to/dir`` that are larger than 10 MB.
+
+Finding files based on modification time::
+
+   find /path/to/dir -mtime +7
+
+This will find all files in ``/path/to/dir`` that have been modified more than 7 days ago.
+
+Finding files based on type::
+
+   find /path/to/dir -type f
+
+This will find all files in ``/path/to/dir`` that are regular files (not directories).
+
+Finding files based on name::
+
+   find /path/to/dir -name "*.txt"
+
+This will find all files in ``/path/to/dir`` that have a ``.txt`` file extension.
+
+Executing commands on matching files::
+
+   find /path/to/dir -name "*.txt" -exec chmod 644 {} \;
+
+This will find all files in ``/path/to/dir`` that have a ``.txt`` file extension and execute the ``chmod`` command on each file, changing its permissions to ``644``.
+
```

Comment:
You might want to include more information about what chmod is and what does the permission 644 do because some people might not be familiar with this command.